### ISSUE TITLE
Review #1

### ISSUE THREAD

**Issue by distillpub-reviewers** [andrefaraujo/receptive_field_paper#6]
_The following peer review was solicited as part of the Distill review process._

_**The reviewer chose to waive anonymity.** Distill offers reviewers a choice between anonymous review and offering reviews under their name. Non-anonymous review allows reviewers to get credit for the service them offer to the community._

_Distill is grateful to the reviewer, The-Hien Dang-Ha, for taking the time to write such a thorough review._

---

The paper devises a nice way of formalizing the calculation of receptive field for both single-path and multi-path networks. Many effective connections are made to relate various equations in a meaningful way. For example, the introduction of "effective stride" and "effective padding" are exceptionally good, which conceptually connect the recurrence equations and their closed-form solutions. The awareness of "effective receptive field" and the need for better understanding of this related concept is appreciated.

The Appendix showing how the recurrence equations are solved, and the open-source library contributing directly to the Tensorflow repository prove the correctness and integrity of this paper.

The figure comparing the receptive field size and the model accuracy could be improved by also embedding the model computation cost (FLOPS) (e.g., as size of the circle), since you acknowledge the fact that the FLOPS dimension is important.

The contribution of the paper is significant, but I would love to see more about the effective receptive field, which I believe to be even more important. Specifically, I would love to see how does the importance of each feature in a receptive field are distributed, before and after training. I might be able to help if you want to dive into this topic.

---

_Distill employs a [reviewer worksheet](https://docs.google.com/document/d/16BNLdSmoUc1zqydECJ8trYkVNDnKVlxT57EEXhCxWSc/edit) as a help for reviewers._

_The first three parts of this worksheet ask reviewers to rate a submission along certain dimensions on a scale from 1 to 5. While the scale meaning is consistently "higher is better", please [read the explanations for our expectations for each score](https://docs.google.com/document/d/16BNLdSmoUc1zqydECJ8trYkVNDnKVlxT57EEXhCxWSc/edit)—we do not expect even exceptionally good papers to receive a perfect score in every category, and expect most papers to be around a 3 in most categories._

_Any concerns or conflicts of interest that you are aware of?_: No known conflicts of interest
_What type of contributions does this article make?_: Explanation of existing results

Advancing the Dialogue | Score
-- | --
How significant are these contributions? | 4/5

Outstanding Communication | Score
-- | --
Article Structure | 4/5
Writing Style | 4/5
Diagram & Interface Style | 3/5
Impact of diagrams / interfaces / tools for thought? | 3/5
Readability | 4/5

Scientific Correctness & Integrity | Score
-- | --
Are claims in the article well supported? | 4/5
Does the article critically evaluate its limitations? How easily would a lay person understand them? | 2/5
How easy would it be to replicate (or falsify) the results? | 5/5
Does the article cite relevant work? | 3/5
Does the article exhibit strong intellectual honesty and scientific hygiene? | 4/5

**Comment by andrefaraujo** [comment]
Thanks The-Hien for the review! 

We will wait for the other reviews to come in and address all of them together, to avoid modifying the online manuscript while other reviewers are working on it. We will follow-up with a detailed response at that point.

**Comment by andrefaraujo** [comment]
Thanks again The-Hien for the review! Please see below our responses to your specific comments/requests:

> The figure comparing the receptive field size and the model accuracy could be improved by also embedding the model computation cost (FLOPS) (e.g., as size of the circle), since you acknowledge the fact that the FLOPS dimension is important.

Thanks for this great suggestion. We have added the number of FLOPs as the size of the circle in the last figure as suggested. To make sure our results are reproducible, and to provide a reference for the different models analyzed in our paper, we have also checked-in the code that computes the number of FLOPs for all models and end-points in the tf.contrib.receptive_field codebase (see [this commit](https://github.com/tensorflow/tensorflow/commit/65b30642f8c5d0fb0b780107a70a4ebbc5938bfa?diff=split)). A comprehensive table including the new results with FLOPs can be found [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/receptive_field/RECEPTIVE_FIELD_TABLE.md).

> I would love to see more about the effective receptive field, which I believe to be even more important. Specifically, I would love to see how does the importance of each feature in a receptive field are distributed, before and after training. I might be able to help if you want to dive into this topic.

We agree that the effective receptive field is very important, and would be very interested to explore this in future work. We believe that this is beyond the scope of this paper, though. Thanks for offering to help! -- we are happy to chat offline once we are done finalizing this paper.

**Comment by andrefaraujo** [comment]
Since we have addressed the suggestions from the reviewer, I am closing this issue now. But please feel free to reopen if you have any further questions/suggestions/comments :)
